example-application/libraries/users-service.js

```diff
@@ -0,0 +1,35 @@
+const axios = require('axios');
+
+module.exports = {
+  async getUser(userId) {
+    let response = await axios.get(
+      `http://localhost/user/${userId}`,
+      {
+        validateStatus: false,
+        timeout: 2000,
+      }
+    );
+
+    console.log(
+      `Asked to get user and get response with status ${response.status}`
+    );
+
+    if (response.status === 503) {
```

Comment:
Nice code, but maybe not simple enough for demo. Maybe it could be simpler to read with:

https://www.npmjs.com/package/axios-retry